src/sparseml/modifiers/obcq/utils/helpers.py

```diff
@@ -22,22 +23,33 @@
 
 
 class Catcher(torch.nn.Module):
-    def __init__(self, module, target_keys):
+    def __init__(self, module):
         super().__init__()
         self.module = module
-        self.cache = {key: [] for key in target_keys}
-        self.target_keys = target_keys
+        self.target_keys = None
+        self.cache = defaultdict(list)
         self.cache["inputs"] = []
 
     def forward(self, *args, **kwargs):
         self.cache["inputs"].append(args)
+        if self.target_keys is None:
+            self.target_keys = self._get_target_keys(kwargs.keys())
+
         for key in self.target_keys:
             self.cache[key].append(kwargs[key])
         raise ValueError
 
     def get_cache(self):
         return self.cache
 
+    def _get_target_keys(self, input_keys):
+        target_keys = []
+        if "attention_mask" in input_keys:
+            target_keys.append("attention_mask")
+        if "position_ids" in input_keys:
+            target_keys.append("position_ids")
```

Comment:
nit: hardcoded constants - use enum or define constants

Reply:
Went with the route recommended by @anmarques; and declared `_DEFAULT_TARGET_IDS` at the top

I didn't want to spend too much time on this for the following reasons:
- right now we do not support targeting `LMhead` for pruning (which we should)
- LMHead has slightly different inputs, so the logic here would need to change eventually to support that
- This UX change is just for the meantime